modules/end-user-guide/pages/selecting-an-ide.adoc

```diff
@@ -5,13 +5,129 @@
 [id="selecting-an-ide_{context}"]
 = Selecting an IDE
 
-You can select your IDE at when starting a workspace: 
+You can select your preferred IDE when starting a workspace: 
+
+* link:https://theia-ide.org/docs/[Eclipse Theia], the current default editor
+* JetBrains IDE such as link:https://www.jetbrains.com/help/idea/getting-started.html[IntelliJ IDEA] or link:https://www.jetbrains.com/help/pycharm/quick-start-guide.html[PyCharm]
+* Visual Studio Code - link:https://github.com/Microsoft/vscode/[Open Source ("Code - OSS")]
+
+Currently, the latest stable Theia is the default editor when none is specified. 
+
+There are multiple ways to select an IDE when launching a workspace:
+
+* using the URL parameter `che-editor`
+* using a `.che/che-editor.yaml` file
+
+
+== Using URL parameters
+
+When launching a workspace using the factory API, you can pass in additional parameters so the chosen IDE is loaded.
+
+```
+https://<che_fqdn>#<git_repository_url>?che-editor=<editor_key>
+```
+
+See the `<editor_key>` values for the supported IDEs in xref:optional-parameters-for-the-urls-for-starting-a-new-workspace.adoc[Table 1. The <editor_key> values for the supported IDEs].
+
+
+== Using a {prod-short} Editor file 
+
+To define a default editor for your project, different from the default in {prod-short}, place a `che-editor.yaml` file in a `.che` folder in the root of your project, and point it at the ID of the editor of choice.
+
+```
+id: che-incubator/che-idea/latest
+```
+
+Ref: https://github.com/che-samples/gradle-demo-project/blob/devfilev2/.che/che-editor.yaml
+
+For a list of supported IDs, see xref:optional-parameters-for-the-urls-for-starting-a-new-workspace.adoc[Optional parameters for the URLs for starting a new workspace].
+
+Experimental new IDEs can be loaded from the default plug-in registry, using the IDs shown in link:https://github.com/eclipse-che/che-plugin-registry/blob/main/che-editors.yaml[che-editors.yaml].
+
+
+
+=== Using a custom plug-in registry for your own editors
+
+If you want to include different editors than the default list in the {prod-short} plug-in registry, you can do so in your `che-editor.yaml` file by setting an optional `registryUrl` directive.
+
+```
+id: eclipse/che-theia/next            # mandatory
+registryUrl: https://my-registry.com  # optional
+override:                             # optional
+  containers:
+    - name: theia-ide
+      memoryLimit: 1280Mi
+```
+
+
+=== Using a web reference
+
+Rather than a reference to a plug-in in your plug-in registry, you can point at a YAML file directly with the `reference` directive.
```

Comment:
Rather than a reference to a plug-in in your plug-in registry, you can point at a YAML file directly with the `reference` directive: